src/components/Question.jsx

```diff
@@ -0,0 +1,47 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import Button from './Button';
+
+const Question = ({ id, question, answers, rightAnswer }) => {
+  const [isCorrect, setIsCorrect] = useState(null);
+
+  const handleOptionChange = (event) => {
+    const correctAnswer = event === rightAnswer;
+
+    setIsCorrect(correctAnswer);
+  };
+
+  const buttonColor = isCorrect ? 'focus:bg-green-500' : 'focus:bg-red-500';
+
+  // const disableButtons = isCorrect !== null;
+
+  return (
+    <div className="py-7">
+      <h1 className="text-3xl first:mt-0 mt-10 w-fit m-auto">
+        <span className="mr-1">{id}.</span>
+        {question}
+      </h1>
+      <div className="mt-6 grid lg:grid-cols-2 sm:grid-cols-1 gap-5">
+        {answers.map((answer) => (
+          <Button
+            id={answer.id}
+            title={answer.title}
+            key={answer.id}
+            handleClick={() => handleOptionChange(answer.id)}
+            className={`${buttonColor}`}
+          />
```

Comment:
to check if disabled need to sutisfy both conditions 
1. ! isCorrect
2. check if correct answer !== answer.id

Reply:
1. add useContext to transfer score between componnets 
2. remove isCorrect and add answer
3. clean previous answer
4. remove logic from Question